client/driver/executor/test_harness_test.go

```diff
@@ -14,6 +15,57 @@ import (
 	"github.com/hashicorp/nomad/nomad/structs"
 )
 
+func TestMain(m *testing.M) {
+	switch os.Getenv("TEST_MAIN") {
+	case "app":
+		appMain()
+	default:
+		os.Exit(m.Run())
+	}
+}
+
+func appMain() {
```

Comment:
We should be able to move almost all of the TestMain/appMain code into a helper package. We will still need a minimal TestMain in each package that needs appMain, but it can get all of its functionality by calling another function like so:

```go
func TestMain(m *testing.M) {
    testhelp.Run(m)
}
```

Where `testhelp.Run` contains the `switch os.Getenv ...` code.